cpp/src/round/round.cu

```diff
@@ -31,6 +33,9 @@ namespace {  // anonymous
 float __device__ generic_round(float f) { return roundf(f); }
 double __device__ generic_round(double d) { return ::round(d); }
 
+float __device__ generic_round_half_even(float f) { return rintf(f); }
```

Comment:
I think the rounding of `rint` depends on a setting specified using `fesetround()`.  So if you depend on rint, don't you need to ensure the correct setting?

Reply:
I have to look into this.

Reply:
[CUDA Math API](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__DOUBLE.html#group__CUDA__MATH__DOUBLE_1g3b8026edb2f2e441669845f0f3fa3bf7 ) docs don't seem to mention it and the intrinsics seem to have a [bunch of "round-to-nearest-even mode"](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INTRINSIC__DOUBLE.html#group__CUDA__MATH__INTRINSIC__DOUBLE) so I wouldn't be surprised it it just works on top of that.

Reply:
Actually the docs do mention it is round to nearest even. 
https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__DOUBLE.html#group__CUDA__MATH__DOUBLE_1g3b8026edb2f2e441669845f0f3fa3bf7

> Round x to the nearest integer value in floating-point format, with halfway cases rounded to the nearest even integer value.

Reply:
Sorry, should have been more clear. When I said
> docs don't seem to mention it

I was referring to `fesetround()`